### PR TITLE
Support with_leading_visual_svg in Primer::Beta::Button

### PR DESCRIPTION
**Pull request by mjimenez98**
_Authors: Please fill out this form carefully and completely._

_Reviewers: By approving this Pull Request you are approving the code change, as well as its deployment and mitigation plans._
_Please read this description carefully. If you feel there is anything unclear or missing, please ask for updates._

### What are you trying to accomplish?

Just like `ActionList` currently supports, I would like to have a SVG serve as the leading visual icon in a `Primer::Beta::Button` component. At the moment only Octicons are allowed.

### Screenshots

Picture of the button component with a SVG as a leading icon:

Screenshot 2023-08-18 at 13 04 37

Also, before/after pictures from changes to Lookbook:

| Before | After |
|--------|--------|
| Screenshot 2023-08-18 at 12 08 58 | Screenshot 2023-08-18 at 12 21 29 | 

#### List the issues that this change affects.

Closes https://github.com/github/primer/issues/2565

#### Risk Assessment
  

- [X] **Low risk** the change is small, highly observable, and easily rolled back.
- [ ] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### What approach did you choose and why?

Extend the `leading_visual` function in `Primer::Beta::Button` to support SVGs. I used the same pattern found in `ActionList`.

Here is the generated HTML for the example found in the Lookbook:

```html
<button id="button-with-leading-svg-preview" type="button" data-view-component="true" class="Button--invisible Button--medium Button">    <span class="Button-content">
    <span class="Button-visual Button-leadingVisual">
      <svg width="16" height="16" data-view-component="true">    <path d="M8 16a2 2 0 001.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 008 16z"></path><path fill-rule="evenodd" d="M8 1.5A3.5 3.5 0 004.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.018.018 0 00-.003.01l.001.006c0 .002.002.004.004.006a.017.017 0 00.006.004l.007.001h10.964l.007-.001a.016.016 0 00.006-.004.016.016 0 00.004-.006l.001-.007a.017.017 0 00-.003-.01l-1.703-2.554a1.75 1.75 0 01-.294-.97V5A3.5 3.5 0 008 1.5zM3 5a5 5 0 0110 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.518 1.518 0 0113.482 13H2.518a1.518 1.518 0 01-1.263-2.36l1.703-2.554A.25.25 0 003 7.947V5z"></path>
      </svg>
    </span>
    <span class="Button-label">Leading SVG button</span>
  </span>
</button>
```

### Anything you want to highlight for special attention from reviewers?

N/A

### Accessibility

- **No new axe scan violation** - This change does not introduce any new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violations.

### Merge checklist

- [X] Added/updated tests
- [X] Added/updated documentation
- [X] Added/updated previews (Lookbook)
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [X] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: previews/primer/beta/button_preview/leading_visual.html.erb [hunk 1]
```diff
@@ -2,9 +2,22 @@
...
-    id: id,
+    id: "button-with-leading-icon-preview",
     align_content: align_content
 )) do |component| %>
   <% component.with_leading_visual_icon(icon: :search) %>
-  Button
+  Leading icon button
+<% end %>
+
+<%= render(Primer::Beta::Button.new(
```
Comment (langermank) [inline previews/primer/beta/button_preview/leading_visual.html.erb:12]: Would you mind actually creating a _new_ preview for the SVG version so we can isolate the behavior of each? 🙌

**Comment by langermank** [comment]
One thing I'm noticing is that a custom SVG won't necessarily access the correct colors defined by the `Button` component CSS. Octicons have `currentColor` set as fill so it will inherit, but in the example SVG you used it doesn't have that. I see two paths:

- Update the CSS to target any `svg` elements inside the button visual class
- Assume all custom SVGs will set the fill to `currentColor`.

What do you think?

**Comment by mjimenez98** [comment]
@langermank Great catch! Thank you!

I like the idea of matching the functionality of `leading_with_icon_svg` to that of `leading_with_icon` 👍 

> Update the CSS to target any svg elements inside the button visual class

I gave this a try in [b7aedb4](https://github.com/primer/view_components/pull/2215/commits/b7aedb4be34868a483073827575847b047e9e1e7). Let me know what you think!

I noticed that the same issue was happening with `ActionList` so I also tried to fix the color issue for it in [b7aedb4](https://github.com/primer/view_components/pull/2215/commits/b7aedb4be34868a483073827575847b047e9e1e7)

File: previews/primer/beta/button_preview/leading_visual.html.erb (hunk 1 above)
Comment (mjimenez98) [reply previews/primer/beta/button_preview/leading_visual.html.erb]: For sure! Done in [e6dd38f](https://github.com/primer/view_components/pull/2215/commits/e6dd38f6c17021ae0a405c71e275eb09ab3c491e)

**Review by jonrohan — APPROVED** [review]
This looks good! Thanks for the contribution ✨

**Review by langermank — APPROVED** [review]
Looks great, thanks! 🚀

**Pull request merged**